src/itype.h

```diff
@@ -607,11 +607,18 @@ struct itype {
 
 // Includes food drink and drugs
 struct it_comest : itype {
+    friend class Item_factory;
```

Comment:
I went with friend class because currently all `itype`s use only the default constructor and because `itype` (the base class) already has `Item_factory` as a friend. It looked like a convention, so I tried not to break it.

I could add a setter or make the field public as it was before. Constructor alone would work now, but it would require either a setter or public field later if we wanted to add mod override for individual item fields (as opposed to overriding the entire item at once).